openam-rest/src/main/java/org/forgerock/openam/cors/CORSService.java

```diff
@@ -148,6 +155,22 @@ public boolean handleRequest(final HttpServletRequest req, final HttpServletResp
 
     }
 
+    /**
+     * Handles a failed CORS (Cross-Origin Resource Sharing) request by generating a standardized
+     * JSON error response and setting the appropriate HTTP status code.
+     *
+     * @param res the {@link HttpServletResponse} to which the error response will be written
+     * @throws IOException if an I/O error occurs while writing the response
+     */
+    private void handleFailedCORS(HttpServletResponse res) throws IOException {
+        ResourceException resourceException = ResourceException.getException(HttpServletResponse.SC_BAD_REQUEST, "CORS error occurred");
+        JsonValue jsonValue = resourceException.toJsonValue();
+        res.setContentType("application/json");
+        res.setCharacterEncoding("UTF-8");
+        res.getWriter().write(jsonValue.toString());
```

Comment:
Just keep in mind that JsonValue#toString doesn't always return actual proper JSON. If you call it on something like a Map value (result of json.get(field)), it will just print out the "value", not a valid JSON.